pyvo/dal/adhoc.py

```diff
@@ -233,14 +233,14 @@ class DatalinkRecordMixin:
 
     - ``getdataset()`` considers datalink.
     """
-    def getdatalink(self):
+    def getdatalink(self, session=None):
         try:
             datalink = self._results.get_adhocservice_by_ivoid(DATALINK_IVOID)
 
-            query = DatalinkQuery.from_resource(self, datalink)
+            query = DatalinkQuery.from_resource(self, datalink, session=session)
```

Comment:
`getdataset` below uses self._session, so can that be use here too? So in the example stated in issue #292, instead of passing `session` twice, once to `SCSService` and once to `getdatalink `, one can do:
```python
query = DatalinkQuery.from_resource(self, datalink, session=self._session)
```